File: Application/Backend/SkuciSeAPI/SkuciSeAPI/Models/Reservation.cs
using SkuciSeAPI.Models.HelperModels;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace SkuciSeAPI.Models
{
    public class Reservation
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }
        public DateTime DateOfReservation { get; set; }
        public Status Status { get; set; }

        // Foreign keys
        public long AdvertId { get; set; }
        public Advert Advert { get; set; }

        public long OwnerId { get; set; }

        public long UserId { get; set; }
        public User User { get; set; }

        public long? CommentId { get; set; }
        public Comment Comment { get; set; }
    }
}

File: Application/SkuciSe/app/src/main/java/com/example/skucise/models/Reservation.kt
package com.example.skucise.models

import com.example.skucise.models.HelperModels.Status
import org.joda.time.DateTime
import org.joda.time.LocalDateTime
import java.io.Serializable

class Reservation (
    var id: Long?,
    var advertId: Long?,
    var advert: Advert?,
    var ownerId: Long?,
    var userId: Long?,
    var user: User?,
    var dateOfReservation: String?,
    var date: LocalDateTime?,
    var status: Status?,
    var commentId: Long?,
    var comment: Comment? = null
): Serializable {}